code/frameworks/svelte-vite/src/plugins/ts2doc.ts

```diff
@@ -0,0 +1,209 @@
+  const shimContent = ts.sys.readFile(shimFilename) || '';
+  const shimSourceFile = ts.createSourceFile(
+    shimFilename,
+    shimContent,
+    ts.ScriptTarget.Latest,
+    true
+  );
+
+  const tsx = svelte2tsx(fileContent, {
+    version: VERSION,
+    isTsFile: true,
+    mode: 'dts',
+  });
+  const currentSourceFile = ts.createSourceFile('tmp.ts', tsx.code, ts.ScriptTarget.Latest, true);
+
+  const host = ts.createCompilerHost({});
+  host.getSourceFile = (fileName, languageVersion, onError) => {
+    if (fileName === 'tmp.ts') {
+      return currentSourceFile;
+    } else if (fileName === shimContent) {
+      return shimSourceFile;
+    } else {
+      // ignore other files
+      return;
+    }
+  };
+
+  // Create a program with the custom compiler host
+  const program = ts.createProgram([currentSourceFile.fileName, shimSourceFile.fileName], {}, host);
+  const checker = program.getTypeChecker();
+
+  const propMap: Map<string, SvelteDataItem> = new Map();
+
+  const renderFunction = currentSourceFile.statements.find((statement) => {
+    return ts.isFunctionDeclaration(statement) && statement.name?.text === 'render';
+  }) as ts.FunctionDeclaration | undefined;
+  if (renderFunction === undefined) {
+    return {
+      runeUsed: false,
+      data: [],
+    };
+  }
+
+  function getPropsFromTypeLiteral(type: ts.TypeLiteralNode) {
+    const members = type.members;
+
+    members.forEach((member) => {
+      if (ts.isPropertySignature(member)) {
+        const name = member.name.getText();
+        let typeString = '';
+        if (member.type !== undefined) {
+          const memberType = checker.getTypeFromTypeNode(member.type);
+          typeString = checker.typeToString(memberType);
+        }
+        const jsDoc = ts.getJSDocCommentsAndTags(member);
+        const docComments = jsDoc
+          .map((doc) => {
+            let s = ts.getTextOfJSDocComment(doc.comment) || '';
+            doc.forEachChild((child) => {
+              // Type information from JSDoc comment
+              if (ts.isJSDocTypeTag(child)) {
+                let t = '';
+                child.typeExpression.forEachChild((ty) => {
+                  t += ty.getText();
+                });
+                if (t.length > 0) {
+                  typeString = t;
+                }
+                s += ts.getTextOfJSDocComment(child.comment);
+              }
+            });
+            return s;
+          })
+          .join('\n');
+
+        // mimic the structure of sveltedoc-parser.
+        propMap.set(name, {
+          name: name,
+          visibility: 'public',
+          description: docComments,
+          keywords: [],
+          kind: 'let',
+          type: { kind: 'type', text: typeString, type: typeString },
+          static: false,
+          readonly: false,
+          importPath: undefined,
+          originalName: undefined,
+          localName: undefined,
+          defaultValue: undefined,
+        });
```

Comment:
if we indeed are able to ditch `sveltedoc-parser` completely, we don't have to mimic its structure and I think we should have whatever structure would make sense in a world where `sveltedoc-parser` wasn't a consideration.

Reply:
But I also see that this PR only touches Vite and not Webpack, and we'd probably need to have the same structure across the two, at least until we can do a breaking change.